lib/graphql/input_object_type.rb

```diff
@@ -57,18 +57,25 @@ def kind
     def validate_non_null_input(input, warden)
       result = GraphQL::Query::InputValidationResult.new
 
+      input_h = input.is_a?(Hash) ? input : (input.to_h rescue nil)
+
+      if input_h.nil?
+        result.add_problem("Expected #{JSON.generate(input, quirks_mode: true)} to be a hash")
```

Comment:
can we say `be a Hash or respond to #to_h`? That way, if someone gets here, they might find a way to help themselves.